Example/Stripe iOS Example (Custom)/SofortExampleViewController.m

```diff
@@ -79,31 +95,30 @@ - (void)pay {
-        [self dismissPollingUI];
-        if (error) {
-            [self.delegate exampleViewController:self didFinishWithError:error];
-        } else {
-            if (source.status == STPSourceStatusConsumed) {
-                [self.delegate exampleViewController:self didFinishWithMessage:@"Payment successfully created"];
-            } else if (source.status == STPSourceStatusFailed) {
-                [self.delegate exampleViewController:self didFinishWithMessage:@"Payment failed"];
-            } else {
-                [self.delegate exampleViewController:self didFinishWithMessage:@"Order received"];
-            }
+            // In order to use STPRedirectContext, you'll need to set up
+            // your app delegate to forward URLs to the Stripe SDK.
+            // See `[Stripe handleStripeURLCallback:]`
+            STPRedirectContext *redirect = [[STPRedirectContext alloc] initWithSource:source completion:^(NSString *sourceID, NSString *clientSecret, NSError *error) {
+                [[STPAPIClient sharedClient] startPollingSourceWithId:sourceID
+                                                         clientSecret:clientSecret
+                                                              timeout:10
+                                                           completion:^(STPSource *source, NSError *error) {
+                                                               [self updateUIForPaymentInProgress:NO];
+                                                               if (error) {
+                                                                   [self.delegate exampleViewController:self didFinishWithError:error];
+                                                               } else {
+                                                                   if (source.status == STPSourceStatusConsumed ||
+                                                                       source.status == STPSourceStatusChargeable) {
+                                                                       [self.delegate exampleViewController:self didFinishWithMessage:@"Payment successfully created"];
+                                                                   } else if (source.status == STPSourceStatusFailed || source.status == STPSourceStatusCanceled) {
+                                                                       [self.delegate exampleViewController:self didFinishWithMessage:@"Payment failed"];
+                                                                   } else {
+                                                                       [self.delegate exampleViewController:self didFinishWithMessage:@"Order received"];
+                                                                   }
+                                                               }
+                                                           }];
+            }];
+            [redirect startSafariViewControllerRedirectFlowFromViewController:self];
```

Comment:
Not sure we want to encourage people to not retain a reference to this context, although I think it might work currently because it retains itself for the length of the redirect.